### ISSUE TITLE
Warn on presence of too many indirections

### ISSUE THREAD

**Issue by chavacava** [mgechev/revive#129]
Expressions that have too many indirections are hard to understand.
For example (from Kubernetes):

```go
o.s.GA.ZoneOperations.Get(o.projectID, o.key.Zone, o.key.Name).Context(ctx).Do()
f.Signature.Receiver.Elem.Name
new(RouteBuilder).typeNameHandler(w.typeNameHandleFunc).servicePath(w.rootPath).Method("DELETE").Path(subPath)
```
Other than hard to read, this kind of code is fragile (calls and slice accesses are chained without checking for nil returns nor empty slices)

I propose to create a rule that warns when an expression has too many indirections (_too many_ must be configurable)

**Comment by mgechev** [comment]
This sounds a bit like the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter). Makes sense to me.

**Comment by chavacava** [comment]
Yes, I've did not mentioned Law of Demeter because a rule checking it is fare more complicated than just calculating call chains length. For example a builder like `MyObject().WithOption1().WithOption2()` chains calls but it respects the Law of Demeter. To detect cases like that we need to typecheck and then control on the types in the call chain.
Most of the time, a simple call chain length is enough to point out _aberrations_ like the ones I've used as example in the proposal.
